src/main/scala/com/nec/spark/planning/LibLocation.scala

```diff
@@ -0,0 +1,38 @@
+package com.nec.spark.planning
+
+import java.nio.file.{Files, Path, Paths, StandardOpenOption}
+
+import com.nec.spark.{RequestCompiledLibraryForCode, RequestCompiledLibraryResponse, SparkCycloneExecutorPlugin}
+
+import org.apache.spark.api.plugin.PluginContext
+
+object LibLocation {
+  trait LibLocation {
+    def resolveLocation(): Path
+  }
+
+  case class LocalLibLocation(libraryPath: String) extends LibLocation {
+    override def resolveLocation(): Path ={
+      if(Files.exists(Paths.get(libraryPath))) {
+        throw new RuntimeException(s"Expected library to be present in ${libraryPath}, but file does not exist")
+      } else {
+        Paths.get(libraryPath).toAbsolutePath
+      }
+    }
+  }
+
+  case class DistributedLibLocation(libraryPath: String, code: String) extends LibLocation {
```

Comment:
We might not want to include the `code` as this makes plans hard to read. Code has should be ok, but in all actuality, `libraryPath` should suffice completely too.

Reply:
Yeah, I used that because it makes sense to have some separation here, instead  of requesting direct path from master. We can change that ofc, but if anything I'd probably prefer passing `CodeLines`to make plans easier to read instead just `libraryPath` but happy to discuss ;))